port/common/omrcuda.cpp

```diff
@@ -292,84 +292,200 @@ J9CUDA_STATIC_ASSERT(
+		return J9CUDA_ERROR_LAUNCH_TIMEOUT;
 	case CUDA_ERROR_NO_BINARY_FOR_GPU:
-		return cudaErrorNoKernelImageForDevice;
+		return J9CUDA_ERROR_NO_KERNEL_IMAGE_FOR_DEVICE;
 	case CUDA_ERROR_NO_DEVICE:
-		return cudaErrorNoDevice;
+		return J9CUDA_ERROR_NO_DEVICE;
 	case CUDA_ERROR_NOT_PERMITTED:
-		return cudaErrorNotPermitted;
+		return J9CUDA_ERROR_NOT_PERMITTED;
 	case CUDA_ERROR_NOT_READY:
-		return cudaErrorNotReady;
+		return J9CUDA_ERROR_NOT_READY;
 	case CUDA_ERROR_NOT_SUPPORTED:
-		return cudaErrorNotSupported;
+		return J9CUDA_ERROR_NOT_SUPPORTED;
 	case CUDA_ERROR_OPERATING_SYSTEM:
-		return cudaErrorOperatingSystem;
+		return J9CUDA_ERROR_OPERATING_SYSTEM;
 	case CUDA_ERROR_OUT_OF_MEMORY:
-		return cudaErrorMemoryAllocation;
+		return J9CUDA_ERROR_MEMORY_ALLOCATION;
 	case CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED:
-		return cudaErrorPeerAccessAlreadyEnabled;
+		return J9CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED;
 	case CUDA_ERROR_PEER_ACCESS_NOT_ENABLED:
-		return cudaErrorPeerAccessNotEnabled;
+		return J9CUDA_ERROR_PEER_ACCESS_NOT_ENABLED;
 	case CUDA_ERROR_PEER_ACCESS_UNSUPPORTED:
-		return cudaErrorPeerAccessUnsupported;
-	case CUDA_ERROR_PROFILER_ALREADY_STARTED:
-		return cudaErrorProfilerAlreadyStarted;
-	case CUDA_ERROR_PROFILER_ALREADY_STOPPED:
-		return cudaErrorProfilerAlreadyStopped;
+		return J9CUDA_ERROR_PEER_ACCESS_UNSUPPORTED;
 	case CUDA_ERROR_PROFILER_DISABLED:
-		return cudaErrorProfilerDisabled;
-	case CUDA_ERROR_PROFILER_NOT_INITIALIZED:
-		return cudaErrorProfilerNotInitialized;
+		return J9CUDA_ERROR_PROFILER_DISABLED;
 	case CUDA_ERROR_SHARED_OBJECT_INIT_FAILED:
-		return cudaErrorSharedObjectInitFailed;
+		return J9CUDA_ERROR_SHARED_OBJECT_INIT_FAILED;
 	case CUDA_ERROR_SHARED_OBJECT_SYMBOL_NOT_FOUND:
-		return cudaErrorSharedObjectSymbolNotFound;
+		return J9CUDA_ERROR_SHARED_OBJECT_SYMBOL_NOT_FOUND;
 	case CUDA_ERROR_TOO_MANY_PEERS:
-		return cudaErrorTooManyPeers;
+		return J9CUDA_ERROR_TOO_MANY_PEERS;
 	case CUDA_ERROR_UNKNOWN:
-		return cudaErrorUnknown;
+		return J9CUDA_ERROR_UNKNOWN;
 	case CUDA_ERROR_UNMAP_FAILED:
-		return cudaErrorUnmapBufferObjectFailed;
+		return J9CUDA_ERROR_UNMAP_BUFFER_OBJECT_FAILED;
 	case CUDA_ERROR_UNSUPPORTED_LIMIT:
-		return cudaErrorUnsupportedLimit;
+		return J9CUDA_ERROR_UNSUPPORTED_LIMIT;
 
 	default:
-		return (cudaError_t)-(int32_t)result;
+		return (J9CudaError)-(int32_t)result;
+	}
+}
+
+struct RuntimeCodePair
+{
+	J9CudaError omrCode;
+	int32_t cudaCode;
+};
+
+/*
+ * Mappings between OMR codes and NVIDIA runtime error codes for version 10.1 and later.
+ */
+const RuntimeCodePair newRuntimeCodePairs[] =
+{
+	/*
+	 * This table was automatically generated by a tool compiled against CUDA 10.1
+	 * header files. The values of 'cudaCode' must be numeric constants because
+	 * they are potentially associated with a version of the runtime which is newer
+	 * than what is available at compile time.
+	 */
+	{ J9CUDA_ERROR_MISSING_CONFIGURATION, 52 },
```

Comment:
It would be nice to document the name of the each CUDA error

Reply:
The mapping is _almost_ obvious: for example, `J9CUDA_ERROR_MISSING_CONFIGURATION` corresponds to `cudaErrorMissingConfiguration`.

Reply:
As long as someone can easily figure out the error code, it looks good to me.

Reply:
Both names are now present in the generated switch statements.